src/hooks/useMultiplayerState.ts

```diff
@@ -63,14 +66,98 @@ export function useMultiplayerState(roomId: string) {
 		[],
 	);
 
-	const onMount = useCallback(
-		(app: TldrawApp) => {
-			app.loadRoom(roomId);
-			app.pause();
-			setAppInstance(app);
-		},
-		[roomId],
-	);
+	async function openFromFileSystem(): Promise<null | {
+		fileHandle: FileSystemFileHandle | null;
+		document: TDDocument;
+	}> {
+		const blob = await fileOpen({
+			description: 'Tldraw File',
+			extensions: [`.tldr`],
+			multiple: false,
+		});
+
+		if (!blob) return null;
+
+		const json: string = await new Promise((resolve) => {
```

Comment:
When this is a class then the line 81 to 89 should be placed in a private method with good name what they do. 
Maybe you can outsource it in a additional function placed in scope but unsure.
This file look currently like a script. For me it is hard to follow what is going on. I think the complet file can be more cleaner and following more the best practice from react.